src/consensus/network_event.rs

```diff
@@ -191,40 +217,41 @@ impl Debug for NetworkEvent {
     }
 }
 
-/// The outcome of polling the chain.
-#[derive(Eq, PartialEq, Serialize, Deserialize)]
-pub struct AccumulatedEvent {
-    pub content: AccumulatingEvent,
-    pub elders_change: EldersChange,
+// Change to section elders.
+#[derive(Default, Eq, PartialEq, Serialize, Deserialize)]
+pub struct EldersChange {
```

Comment:
The way I look at it is an struct describing the change itself, that is, which nodes were added and which removed (a diff). So calling it `Change` seems OK? 

But apart from that, I like your suggestion. In the past, this actually contained separate fields for our-section elders and for neighbour elders so the naming was more appropriate then.

Or we can go the other way and make it generic: `SetChange` / `SetDiff` ?

Reply:
We're getting into some interesting details of distinction now :)

> So calling it `Change` seems OK?

This is actually the definition of an event. It is a change (see https://en.wikipedia.org/wiki/Event-driven_architecture). So, I would say we are kind of duplicating the language if we come up with an item called `Change`, that is not an event. We could come up with another one that is `Progress`, yet another one that is `Alteration` and so on.. Or, we identify all these things as events, fully embrace the event driven architecture that is half-way adopted already (and certainly is a very good fit for a decentralized network), and as a result we get a simpler structure/taxonomy.

Now, that said, maybe this `EldersChange` struct is part of something else that today is called an event? But if that event is not only about the elder change, then it seems to me we have bundled up several changes into one change.

*looking into the code*

Okay, so I attribute this impedance mismatch to the half-way adoption of the event driven architecture. So, what I see in the code is that developers have correctly used and identified event concepts, but it can be seen that all over, in details as well as overall, that there hasn't been much or any experience with event driven architectures. So there's a lot of "non-event driven" things mixed in there.

This for example:

```
// Connect to all elders from our section or neighbour sections that we are not yet connected
    // to and disconnect from peers that are no longer elders of neighbour sections.
    fn update_peer_connections(&mut self, core: &mut Core, change: &EldersChange) {
        let our_needed_connections: HashSet<_> = self
            .shared_state
            .known_nodes()
            .map(|node| *node.peer_addr())
            .collect();

        for p2p_node in &change.neighbours_removed {
            // The peer might have been relocated from a neighbour to us - in that case do not
            // disconnect from them.
            if our_needed_connections.contains(p2p_node.peer_addr()) {
                continue;
            }

            core.transport.disconnect(*p2p_node.peer_addr());
        }
    }
```

.. is what a Saga does. It listens for certain events, and sends cmds as a result of it.

So, being explicit with such systematization, actually makes the code simpler. There's not just a loose bunch of different objects leading to different things. There's a clearly defined pattern of flow: cmds become events, events are picked up and lead to new cmds. When we create a structure for that, actual types that do things in a specific way, it is much easier for developers to see the overall structure, as well as where they are in a specific flow at any given point.

> Or we can go the other way and make it generic: `SetChange` / `SetDiff` ?

I would not recommend it. Changes are about system behaviour, making a system event driven, and making events first class citizens, is about putting the changes - the behaviour - at the fore front. That means the events are granular and the event names are specific.

The problem here is deep, so I don't think we can fix it at this specific line of code :)

We have things like `handle_neighbour_info_event` and `handle_section_info_event`, and the "misinterpretation of events", the deviation from the principles I describe, that I consider be pervasive throughout the code, continues all the way. (Clarifying: `neighbour_info_event` is not considered a good naming of an event, it is not describing a change, it is not event centric, it is data centric).

Reply:
OK, will make the changes as you are suggesting. 

Regarding things like `handle_section_info_event` - here "event" just means the thing that comes out of parsec (so `AccumulationEvent`) which we are already discussing elsewhere. So the convention we are currently using is `handle_foo` means handle a received message `Foo`, whereas `handle_foo_event` means handle the consensus/accumulation of an "event" 'Foo`. I admit it's not a particularly great convention. Maybe we should be more explicit:
`handle_foo_message` and `handle_foo_consensus` (or `handle_foo_accumulation`) ?

Reply:
Ok I see.
"handle" is a good description sometimes of what a method does. I use it at very high levels, when there is not much behaviour associated with the handling. Otherwise I try to instead name the method according to _what kind of handling_ is going on. Otherwise we run big risk of having the code littered with "handle_" this and that.

It is kind of implicit (often) that a method taking am event as arg handles an event so the information we're passing along at that point (prefixing with "handle") is not very useful and tend to become noise.

In safe_vaults this is something I have identified and I have started naming methods just straight up and down what the intended behaviour is, instead of a technical description of where something came from, what it is and how it will be handled.
As a very recent example from CH refactoring merged just yesterday:

`handle_[do_something_cmd]_client_req`
became
`initiate_[something to be done]`

and
`handle_[do_something_cmd]_vault_req`
became
`finalize_[something to be done]`

We that way went into describing the behaviour and flow (first cmd initiates, the other is a finalization of such a cmd), instead of the technical details around it.

So, to connect back to this. Could we simply call it `accumulate_foo`, or if it was already accumulated, describe what is actually done as a next step with this accumulated foo?